Added English and German translation files.

emanuelschuetze opened the pull request:
- Use gulp and bower to create client-side translations with angular-gettext.
- Add English po template files for django and angular-gettext.
- Added German translation files.
- Fixed some English strings.

emanuelschuetze committed:
Added English and German translation files.

* Use gulp and bower to create client-side translations with angular-gettext.
* Add English po template files for django and angular-gettext.
* Added German translation files.
* Fixed some English strings.

normanjaeckel commented on .gitignore:
```diff
@@ -7,8 +7,13 @@ __pycache__/
 .virtualenv/
 .venv/
 
-# Local user data (settings, database, media, search index, static files)
```
Please readd these lines. They are required for Plugin development with virtualenv installed OpenSlides (core).

emanuelschuetze replied:
done

normanjaeckel commented on bower.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "openslides-votecollector-plugin",
```
I would prefer `openslides-votecollector`.

emanuelschuetze replied:
done

normanjaeckel commented on package.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "openslides-votecollector-plugin",
```
Here the same.

emanuelschuetze replied:
done

emanuelschuetze committed:
Remarks. Use ugettext_noop.

emanuelschuetze committed:
Fix getVote function to get correct percentages.

emanuelschuetze commented:
@normanjaeckel All done. Ready for review and  merge.

emanuelschuetze committed:
Improved translation messages for keypads.

emanuelschuetze committed:
Important fix for seating plan autoupdate: Use keypad.id instead of keypad.keypad_id

in SlideMotionPollCtrl

Note: keypad.id is referenced by MotionPollKeypadConnection.keypad_id

emanuelschuetze committed:
Allow public (namely) voting also for election.

Added keypad to AssigmnetPollKeypadConnection to show single vote of a user/keypad.
Added assigmentpoll slide with seating plan.
Allow to anonymize.
Refactor code.
A lot of template improvements.

emanuelschuetze committed:
Updated translations

emanuelschuetze committed:
Fix percentage on poll slides

emanuelschuetze commented:
@normanjaeckel Ping. Ready for review and merge.

normanjaeckel commented:
Please strip out all *.mo files and add a pattern for them to the gulpignore file.

emanuelschuetze commented:
Why? We have to commit the German mo file to use the django translations.

emanuelschuetze commented:
Only the English mo file is optional (not for usage.)

normanjaeckel commented on openslides_votecollector/views.py:
```diff
@@ -114,8 +114,10 @@ def anonymize_votes(self, request):
-        MotionPollKeypadConnection.objects.filter(poll_id=request.data.get('poll_id')).update(keypad=None)
-        # TODO: Trigger autoupdate.
+        # Clear keypad id and save model to trigger autoupdate.
+        for apkc in MotionPollKeypadConnection.objects.filter(poll_id=request.data.get('poll_id')):
```
typo: `mpck`

emanuelschuetze replied:
done

normanjaeckel commented on openslides_votecollector/views.py:
```diff
@@ -501,12 +533,12 @@ def post(self, request, item_id, keypad_id):
                 # User is already on the speaker list.
                 pass
-            content = _('Added to list of speakers')
+            content = _('Added to        list of speakers')
```
Why did you add whitespaces?

emanuelschuetze replied:
see below.

normanjaeckel commented on openslides_votecollector/views.py:
```diff
@@ -501,12 +533,12 @@ def post(self, request, item_id, keypad_id):
             # Remove speaker if on "next speakers" list (begin_time=None, end_time=None).
             Speaker.objects.filter(user=keypad.user, item=item, begin_time=None, end_time=None).delete()
-            content = _('Removed from list of speakers')
+            content = _('Removed from    list of speakers')
```
Why did you add whitespaces?

emanuelschuetze replied:
We need the whiespace to get a nice linebreak for the Engage keypad display: 

```
Remove from
list of speakers
```

normanjaeckel commented:
@emanuelschuetze Some remarks.

emanuelschuetze commented:
@normanjaeckel done.

emanuelschuetze committed:
Remark

normanjaeckel commented:
I generally do not want to commit compiled translation files into the repo. For release we can add fresh compilation to the built package. But you can leave them trhere for the moment and we can discuss this thing later.

emanuelschuetze commented:
It standard also for OpenSlides core to commit the mo files, see https://github.com/OpenSlides/OpenSlides/tree/master/openslides/locale/de/LC_MESSAGES
